cluster/node.go

```diff
@@ -255,13 +255,14 @@ func (n *HTTPNode) SetPartitions(part []int32) {
 	n.Updated = time.Now()
 }
 
-func (n HTTPNode) Post(ctx context.Context, name, path string, body Traceable) (ret []byte, err error) {
+func (n HTTPNode) PostRaw(ctx context.Context, name, path string, body Traceable) (io.ReadCloser, error) {
 	ctx, span := tracing.NewSpan(ctx, Tracer, name)
 	tags.SpanKindRPCClient.Set(span)
 	tags.PeerService.Set(span, "metrictank")
 	tags.PeerAddress.Set(span, n.RemoteAddr)
 	tags.PeerHostname.Set(span, n.Name)
 	body.Trace(span)
+	var err error
 	defer func(pre time.Time) {
```

Comment:
Interesting side effect here is that this span now finishes before the request body is completely read (e.g. pretty much as soon as the first header is received). This means that it looks like this step is done before the peer is done streaming. Not sure how much this matters.

Reply:
seems this could get confusing. we may end the span whereas the majority of the body still needs to be streamed over the network - and hence take much more time which is not clear in the tracing UI.

why don't we make it the responsibility of PostRaw's caller to finalize the span after it is done reading the body?
(or we make it the caller's responsibility to wrap PostRaw with all the tracing stuff. but seems fine to keep that boilerplate into the beginning of PostRaw)

Reply:
We could return a type that implements the `io.ReadCloser` interface and also finishes the span when the body is closed.

Reply:
that would only help if we don't return an error.
if we return an error, the caller should still finish the span, but would not read or close the body.

Reply:
The type could handle that logic if needed as well. and then the caller would be expected to always call Close

Reply:
> should this code be considered alpha? are you running this in production?

We have been running this in production for a little over a month.